mantidproject/mantid
# 32093 max ent groups

**Description of work.**


The muon group want to be able to use the groups (e.g. fwd, bwd) for MaxEnt calculations. This PR adds the code needed to allow this. 

This PR also fixes a bug that meant that the phase tables from MuonMaxEnt were not available in phase tab.
**To test:**


Open frequency domain analysis
Load `MUSR 62260`
Go to the transform tab
Change the mode from FFT to MaxEnt
Tick the `Output phase table` option
Click calculate
You should get a nice peak in the plot
In the top table, the phase table row will now have an extra option. Select it
Click calculate and you will get a nicer peak
Go to the grouping tab and unselect/add a group (all that matters is that the number of selected groups changes)
Go back to the transform tab
For the phase table row, there will only be one option "None"
Click calculate 
A new option will appear in the phase table row
Set the number of selected groups back to 4 on the grouping tab
On the transform tab, the phase table row will have the original phase table as an option

Open frequency domain analysis
Load `HIFI 84447`
Go to the transform tab
Change the mode from FFT to MaxEnt
Tick the `Output phase table` option
Click calculate
You should get a nice peak in the plot
In the top table, the phase table row will now have an extra option. Select it
Click calculate and you will get a nicer peak
Use the period selector to change the period
Do another maxent calculation - the result should change

Open frequency domain analysis
Load `HIFI 134028-30`
Go to the transform tab
Change the mode from FFT to MaxEnt
Tick the `Output phase table` option
Click calculate
You should get a nice peak in the plot
Change the workspace selector to the next run
Click calculate - the peak will change

Open frequency domain analysis
Load `MUSR 62260`
Go to the transform tab
Change the mode from FFT to MaxEnt
Tick the `Output phase table` option
Change the 'method' row to All detectors
Click calculate
Check that a phase table has appeared in the phase table row options
Go to the phase tab
You should have a phase table to use for phasequad. 
For the other tests, you will not get a phase table in the phasequad tab (it needs the method to be all detectors)

Fixes #32093. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
